newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js

```diff
@@ -0,0 +1,175 @@
+// @flow
+import * as React from 'react';
+import { type AuthenticatedUser } from '../../Profile/AuthenticatedUserContext';
+import { type FileMetadata } from '..';
+import {
+  commitVersion,
+  createCloudProject,
+  getCredentialsForProject,
+  updateCloudProject,
+} from '../../Utils/GDevelopServices/Project';
+import { serializeToJSON } from '../../Utils/Serializer';
+import { initializeZipJs } from '../../Utils/Zip.js';
+import CloudSaveAsDialog from './CloudSaveAsDialog';
+
+const zipProject = async (project: gdProject) => {
+  const zipJs: ZipJs = await initializeZipJs();
+  const projectJson = serializeToJSON(project);
+  const textReader = new zipJs.TextReader(projectJson);
+
+  return new Promise((resolve, reject) => {
+    zipJs.createWriter(new zipJs.BlobWriter('application/zip'), zipWriter => {
+      zipWriter.add('game.json', textReader, () => {
+        zipWriter.close(blob => {
+          resolve(blob);
+        });
+      });
+    });
+  });
+};
+
+const zipProjectAndCommitVersion = async ({
+  authenticatedUser,
+  project,
+  cloudProjectId,
+}: {
+  authenticatedUser: AuthenticatedUser,
+  project: gdProject,
+  cloudProjectId: string,
+}): Promise<?string> => {
+  const archive = await zipProject(project);
+  const newVersion = await commitVersion(
+    authenticatedUser,
+    cloudProjectId,
+    archive
+  );
+  return newVersion;
+};
+
+export const generateOnSaveProject = (
+  authenticatedUser: AuthenticatedUser
+) => async (project: gdProject, fileMetadata: FileMetadata) => {
+  const newVersion = await zipProjectAndCommitVersion({
+    authenticatedUser,
+    project,
+    cloudProjectId: fileMetadata.fileIdentifier,
+  });
+  if (!newVersion) return { wasSaved: false, fileMetadata };
+  return {
+    wasSaved: true,
+    fileMetadata,
+  };
+};
+
+export const generateOnChangeProjectProperty = (
+  authenticatedUser: AuthenticatedUser
+) => async (
+  project: gdProject,
+  fileMetadata: FileMetadata,
+  properties: { name: string }
+): Promise<boolean> => {
+  if (!authenticatedUser.authenticated) return false;
+  try {
+    await updateCloudProject(
+      authenticatedUser,
+      fileMetadata.fileIdentifier,
+      properties
+    );
+    const newVersion = await zipProjectAndCommitVersion({
```

Comment:
It only saves if the name is changed, and only if it's a cloud stored project so I'm not sure how to proceed with this. I agree with you that it should mention it will save though.

Reply:
Good idea, I added it